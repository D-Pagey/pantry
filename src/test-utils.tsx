import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React, { FC, ReactNode } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { FirebaseContext } from './components/ProviderFirebase';

type AllTheProvidersProps = {
    children: ReactNode;
    customContext?: any;
};

const AllTheProviders: FC<AllTheProvidersProps> = ({ children, customContext }) => (
    <FirebaseContext.Provider
        value={{
            categories: [],
            deleteFoodItem: () => () => null,
            expiringCount: 0,
            isAuthed: false,
            isCheckingAuth: false,
            fridge: [],
            signOut: () => null,
            updateFridge: () => null,
            user: {
                email: null,
                name: null
            },
            ...customContext
        }}
    >
        <MemoryRouter>{children}</MemoryRouter>
    </FirebaseContext.Provider>
);

const customRender = (ui: any, customContext?: any, options?: any) =>
    render(ui, { wrapper: () => <AllTheProviders customContext={customContext}>{ui}</AllTheProviders>, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
