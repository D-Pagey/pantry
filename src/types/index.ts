export interface DatabaseCategoryType {
    colour: string;   
    id: string;
    name: string;
}

export interface CategoryType extends DatabaseCategoryType {
    count: number;
    label: string;
    value: string;
    __isNew__?: boolean;
}

export interface KeyedDatabaseCategoryType {
    [id: string]: CategoryType;
}

export type FoodTypes = {
    categories: string[];
    expires: Date;
    id: string;
    name: string;
    servings: number;
};

export type EditFoodTypes = {
    categories: CategoryType[];
    expires: Date;
    id: string;
    name: string;
    servings: number;
};

export type CategoryCountType = {
    [id: string]: number;
};