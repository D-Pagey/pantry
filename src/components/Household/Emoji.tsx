import React, { FC } from 'react';
import { HouseRoleType } from '../../types';
import * as S from './styles';

type EmojiProps = {
    houseRole: HouseRoleType;
};

export const Emoji: FC<EmojiProps> = ({ houseRole }) => {
    if (houseRole === 'admin') {
        return (
            // eslint-disable-next-line
            <S.Span role="img" aria-label="avocado">
                ðŸ¥‘
            </S.Span>
        );
    }

    if (houseRole === 'tenant') {
        return (
            // eslint-disable-next-line
            <S.Span role="img" aria-label="carrot">
                ðŸ¥•
            </S.Span>
        );
    }

    if (houseRole === 'alexa') {
        return (
            // eslint-disable-next-line
            <S.Span role="img" aria-label="robot">
                ðŸ¤–
            </S.Span>
        );
    }

    // pending
    return (
        // eslint-disable-next-line
        <S.Span role="img" aria-label="potato">
            ðŸ¥”
        </S.Span>
    );
};
