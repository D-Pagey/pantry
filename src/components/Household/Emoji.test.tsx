import React from 'react';
import { render, screen } from '../../test-utils';
import { HouseRoleType } from '../../types';
import { Emoji } from './Emoji';

const props = {
    houseRole: 'admin' as HouseRoleType
};

describe('Emoji component', () => {
    it('should render', () => {
        const { container } = render(<Emoji {...props} />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it.each`
        houseRole    | emoji
        ${'admin'}   | ${'ðŸ¥‘'}
        ${'tenant'}  | ${'ðŸ¥•'}
        ${'alexa'}   | ${'ðŸ¤–'}
        ${'pending'} | ${'ðŸ¥”'}
    `('should render $emoji for $houseRole', ({ emoji, houseRole }) => {
        render(<Emoji {...props} houseRole={houseRole} />);

        screen.getByText(emoji);
    });
});
