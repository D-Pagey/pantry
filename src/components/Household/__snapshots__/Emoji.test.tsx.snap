// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoji component should render 1`] = `
.c0 {
  font-size: 22px;
  grid-row: 1 / 3;
  text-align: center;
}

<span
  aria-label="avocado"
  class="c0"
  role="img"
>
  ðŸ¥‘
</span>
`;
